### DISCUSSION TITLE
Use good job gem without uuid

### DISCUSSION THREAD

**Discussion by lethunder** [bensheldon/good_job#1178] [General]
Hey guys,
I'm wondering if it is possible to use good job without uuid ids.i'm developing an app on top of an existing database.i have a lot of contraints.one of this IS that i can't enable pg crypto. Therefore, i can't use uuid.
When i change in migration uuid to id job processor will keep failing before good job main table still expects uuid.
Any Idea?

**Comment by bensheldon** [comment]
You'd need to make sure that all primary IDs and foreign key references are converted. I think it should work (I can't think of anything that specifically in Ruby that validates the type).

What version of Postgres are you using? `pgcrypto` isn't necessary on PG 13+

https://github.com/bensheldon/good_job/blob/1e8954fe5dd181e86326fd4287ed4b2d954ab8d1/lib/generators/good_job/templates/install/migrations/create_good_jobs.rb.erb#L5-L6

**Reply by lethunder** [reply]
That's what i did before posting the question.
look at this
![image](https://github.com/bensheldon/good_job/assets/6217812/e6c0d438-1cae-46b7-a127-7aef25bac315)
looks at the sequence
![image](https://github.com/bensheldon/good_job/assets/6217812/bbf1cada-1b2c-4e29-a73c-60fe75ae3129)

This is my migration file
`class ChangeGoodJobTables < ActiveRecord::Migration[7.1]
  def change
    drop_table :good_jobs, force: :cascade
    create_table :good_jobs do |t|
      t.text :queue_name
      t.integer :priority
      t.jsonb :serialized_params
      t.datetime :scheduled_at
      t.datetime :performed_at
      t.datetime :finished_at
      t.text :error

      t.timestamps

      t.integer :active_job_id
      t.text :concurrency_key
      t.text :cron_key
      t.integer :retried_good_job_id
      t.datetime :cron_at

      t.integer :batch_id
      t.integer :batch_callback_id

      t.boolean :is_discrete
      t.integer :executions_count
      t.text :job_class
      t.integer :error_event, limit: 2
    end

    drop_table :good_job_batches, force: :cascade
    create_table :good_job_batches do |t|
      t.timestamps
      t.text :description
      t.jsonb :serialized_properties
      t.text :on_finish
      t.text :on_success
      t.text :on_discard
      t.text :callback_queue_name
      t.integer :callback_priority
      t.datetime :enqueued_at
      t.datetime :discarded_at
      t.datetime :finished_at
    end

    drop_table :good_job_executions, force: :cascade
    create_table :good_job_executions do |t|
      t.timestamps

      t.integer :active_job_id, null: false
      t.text :job_class
      t.text :queue_name
      t.jsonb :serialized_params
      t.datetime :scheduled_at
      t.datetime :finished_at
      t.text :error
      t.integer :error_event, limit: 2
    end

    drop_table :good_job_processes, force: :cascade
    create_table :good_job_processes do |t|
      t.timestamps
      t.jsonb :state
    end

    drop_table :good_job_settings, force: :cascade
    create_table :good_job_settings do |t|
      t.timestamps
      t.text :key
      t.jsonb :value
      t.index :key, unique: true
    end

    add_index :good_jobs, :scheduled_at, where: "(finished_at IS NULL)", name: "index_good_jobs_on_scheduled_at"
    add_index :good_jobs, [:queue_name, :scheduled_at], where: "(finished_at IS NULL)", name: :index_good_jobs_on_queue_name_and_scheduled_at
    add_index :good_jobs, [:active_job_id, :created_at], name: :index_good_jobs_on_active_job_id_and_created_at
    add_index :good_jobs, :concurrency_key, where: "(finished_at IS NULL)", name: :index_good_jobs_on_concurrency_key_when_unfinished
    add_index :good_jobs, [:cron_key, :created_at], name: :index_good_jobs_on_cron_key_and_created_at
    add_index :good_jobs, [:cron_key, :cron_at], name: :index_good_jobs_on_cron_key_and_cron_at, unique: true
    add_index :good_jobs, [:active_job_id], name: :index_good_jobs_on_active_job_id
    add_index :good_jobs, [:finished_at], where: "retried_good_job_id IS NULL AND finished_at IS NOT NULL", name: :index_good_jobs_jobs_on_finished_at
    add_index :good_jobs, [:priority, :created_at], order: { priority: "DESC NULLS LAST", created_at: :asc },
              where: "finished_at IS NULL", name: :index_good_jobs_jobs_on_priority_created_at_when_unfinished
    add_index :good_jobs, [:batch_id], where: "batch_id IS NOT NULL"
    add_index :good_jobs, [:batch_callback_id], where: "batch_callback_id IS NOT NULL"

    add_index :good_job_executions, [:active_job_id, :created_at], name: :index_good_job_executions_on_active_job_id_and_created_at
  end
end
`

**Reply by bensheldon** [reply]
Oh right! Yep, it won't work because Active Job uses UUIDs internally. 

Hmm, maybe that could probably be monkeypatched around: 

https://github.com/rails/rails/blob/531643244474446efc061b559504ed67acc8cae1/activejob/lib/active_job/core.rb#L95

You're stepping off the beaten path here though.

**Reply by lethunder** [reply]
yeah! thanks
